nav2_util/include/nav2_util/geometry_utils.hpp

```diff
@@ -32,13 +34,23 @@ inline geometry_msgs::msg::Quaternion orientationAroundZAxis(double angle)
   return tf2::toMsg(q);
 }
 
-inline double euclideanDistance(
-  const geometry_msgs::msg::Point pos1,
-  const geometry_msgs::msg::Point pos2)
+inline double euclidean_distance(
+  const geometry_msgs::msg::Point & a,
+  const geometry_msgs::msg::Point & b)
 {
-  const double dx = pos1.x - pos2.x;
-  const double dy = pos1.y - pos2.y;
-  const double dz = pos1.z - pos2.z;
+  double dx = a.x - b.x;
+  double dy = a.y - b.y;
+  double dz = a.z - b.z;
+  return std::sqrt(dx * dx + dy * dy + dz * dz);
```

Comment:
One major reason to use `hypot` over this would be to avoid overflowing. I am not sure if that would be the case here. But as a safety measure I'd suggest `hypot`.

Reply:
`hypot` with 3 parts (x,y,z)  isn't available in STL until C++17, we're on C++11 (and 14 in Foxy)

Reply:
Ohh, my bad!